src/akamai/lib/AkamaiHDPlayer.as

```diff
@@ -406,7 +424,26 @@ package
      * @method AkamaiHDPlayer#totalBitratesAvailable
      */
     public function  totalBitratesAvailable():void
-    { 
+    {
+      if (_bitrateIdArray.length > 0 ) return;
+      var eventObject:Object = new Object();
+      var id:String;
+      if (getStreamsCount() > 0)
+      {
+        for (var i:int = 0; i < getStreamsCount(); i++)
+        {
+          _bitrateIdArray.push((_streamController.mediaPlayer.getBitrateForDynamicStreamIndex(i)) + "kbps");
+          id = _bitrateIdArray[i];
+          var bitrateObject:Object = new Object();
+          bitrateObject.id = id;
+          bitrateObject.height = 0;
+          bitrateObject.width = 0;
+          bitrateObject.bitrate = _streamController.mediaPlayer.getBitrateForDynamicStreamIndex(i) * 1000;
+          _bitrateArray[id] = [ bitrateObject, i];
```

Comment:
This is being used as a map and not an Array, isn't it? The "id" variable is a String so you're not indexing into it, you're setting key value pairs on it as if it were an Object. Array may not be the clearest data type here for _bitrateArray.